class some_exception1 extends Throwable {}
;

class some_exception2 extends some_exception1 {}
;

class catch1 {
  //@ ensures \result == true || \result == false;
  public static boolean fun() {
    //@ ensures (\exists some_exception1 e; e != null; e instanceof some_exception2) ==> \result == true;
    //@ ensures !(\exists some_exception1 e; e != null; e instanceof some_exception2) ==> \result == false;
    try {
      throw new some_exception2();
    } catch (some_exception1 e) {
      if(!(e instanceof some_exception2)) return false;
      if (e instanceof some_exception2) return true;
    }
    return false;
  }
}

